pkg/apis/core/types.go

```diff
@@ -3348,6 +3348,45 @@ type NodeSystemInfo struct {
 	Architecture string
 }
 
+// NodeConfigStatus describes the status of the config assigned by Node.Spec.ConfigSource.
+type NodeConfigStatus struct {
+	// Assigned reports the node's understanding of intended config. When the node detects an
+	// update to Node.Spec.ConfigSource, it updates Node.Status.Config.Assigned before
+	// taking further action. The node tries to make the Assigned config the Active config
+	// by downloading, checkpointing, validating, and attempting to use the referenced source.
+	// +optional
+	Assigned *NodeConfigSource
```

Comment:
TODO: Think through this case more:
1. API server is upgraded to version w/ new source subtype, Kubelets not
2. User sets new source type
3. Kubelet sets `Assigned` in runtime status manager, intending to ack, but had unmarshaled a config with all nil subfields, as far as it could see. So the status sync will fail API server validation.
4. Kubelet sees `AllNilSubfieldsError` when it tries to produce a config source a little after updating runtime status manager, and updates the status manager with this error. But since `Assigned` was already set to an invalid value in the status manager, all status sync attempts will fail API server validation until this is corrected.

Reply:
Now that we report the checkpointed intent in Assigned, rather than using it as an early ack, this isn't a concern. `AllNilSubfieldsError` would be encountered prior to checkpointing the record of Assigned config.